policy/coding-best-practices.md

```diff
@@ -0,0 +1,93 @@
+# Coding Best Practices
+
+This document describes some typical programming pitfalls and best practices
+related to C# and .NET.  It will grow and change as we encounter new situations
+and the codebase evolves.
+
+## Correctness & Business Logic
```

Comment:
I'd add a few bullet points describing how/when to use Debug.Assert, the codebase tends to use this in slightly odd ways - in some cases, to validate data being received from a SQL Server instance. It'd be good to define whether we want to add that to strengthen test coverage, or whether it's being erroneously used as a substitute for throwing an exception. I'm personally leaning slightly towards the latter.